x-pack/build_chromium/README.md

```diff
@@ -1,12 +1,117 @@
 
+This document is an attempt to note all of the gotchas we've come across while building, so that the next time we have to tinker here, we'll have a good starting point.
 
-## Updating the revision
-If you want to build the headless_shell that mirrors a specific version of the Chrome browser, you can take the version number displayed in `Chrome -> About` and plug it into the "Version" input [here](https://omahaproxy.appspot.com/) and it'll give you the Commit that you can feed into the build script using the `--git-sha` argument. It's highly recommended to update the default when you're upgrading the version.
+## Build args
+
+Chromium is built via a build tool called "ninja". The build can be configured by specifying build flags either in an "args.gn" file or via commandline args. We have an "args.gn" file per platform:
+
+- mac: darwin/args.gn
+- linux: linux/args.gn
+- windows: windows/args.gn
+
+The various build flags are not well documented. Some are documented [here](https://www.chromium.org/developers/gn-build-configuration). Some, such as `enable_basic_printing = false`, I only found by poking through 3rd party build scripts.
+
+As of this writing, there is an officially supported headless Chromium build args file for Linux: `build/args/headless.gn`. This does not work on Windows or Mac, so we have taken that as our starting point, and modified it until the Windows / Mac builds succeeded.
+
+## VMs
+
+I ran Linux and Windows VMs in GCP with the following specs:
+
+- 8 core vCPU
+- 30GB RAM
+- 128GB hard drive
+- Ubuntu 18.04 LTS (not minimal)
+- Windows Server 2016 (full, with desktop)
+
+The more cores the better, as the build makes effective use of each. For Linux, Ubuntu is the only officially supported build target.
+
+- Linux:
+  - SSH in using [gcloud](https://cloud.google.com/sdk/)
+  - Get the ssh command in the [GCP console](https://console.cloud.google.com/) -> VM instances -> your-vm-name -> SSH -> gcloud
+  - Their in-browser UI is kinda sluggish, so use the commandline tool
+
+- Windows:
+  - Install Microsoft's Remote Desktop tools
+  - Get the RDP file in the [GCP console](https://console.cloud.google.com/) -> VM instances -> your-vm-name -> RDP -> Download the RDP file
+  - Edit it in Microsoft Remote Desktop:
+    - Display -> Resolution (1280 x 960 or something reasonable)
+    - Local Resources -> Folders, then select the folder(s) you want to share, this is at least `build_chromium` folder
+    - Save
+
+## Initializing each VM / environment
+
+You only need to initialize each environment once.
+
+Create the build folder:
+
+- Mac / Linux: `mkdir -p ~/chromium`
+- Windows: `mkdir c:\chromium`
+
+Copy the `x-pack/build-chromium` folder to each. Replace `you@your-machine` with the correct username and VM name:
+
+- Mac: `cp -r ~/dev/elastic/kibana/x-pack/build_chromium ~/chromium/build_chromium`
```

Comment:
I'd get rid of the `dev` part of the path here, since that depends on where your kibana repo is.  Maybe just start with `kibana`.  Also... the below ones doesn't seem to match up.  Should `~/dev/build_chromium/` be `kibana/x-pack/build_chromium/`?

Reply:
I don't copy the Kibana codebase to the build servers. Just the build_chromium folder. Really, build_chromium should probably be its own repo / project, as it's essentially independent of Kibana. So, it should probably be `cp -r kibana/x-pack/build_chromium ~/chromium/build_chromium`, assuming that you're wd has `kibana` as a child.

Reply:
Ah. You're right. ~dev/build_chromium is wrong.